### ISSUE TITLE
reorganize /dist to simplify CommonJS import

### ISSUE THREAD

**Issue by rodneyrehm** [medialize/ally.js#53]
While discussing ally with @darobin, it became clear that using modules (with systems like browserify) is not ideal yet. We're assuming two things:

1. users will (at least currently) prefer CommonJS modules
2. users of AMD and ES6 can configure paths and are used to it

the current structure is

```
node_modules/ally.js/
├── dist
│   ├── ally.min.js (UMD)
│   ├── ally.min.js.map (UMD)
│   ├── amd
│   │   └── <AMD files>
│   └── common
│       └── <CJS files>
└── src
    └── <ES6 files>
```

the proposed structure is

```
node_modules/ally.js/
├── ally.min.js (UMD)
├── ally.min.js.map (UMD)
├── <CJS files>
├── amd
│   └── <AMD files>
└── src
    └── <ES6 files>
```

which would allow the following - without any configuration - in CommonJS based systems:

```js
// load ally.min.js UMD bundle
var bundle = require('ally.js');
// load the same structure as the UMD bundle, but from CommonJS modules
var bundleFromModules = require('ally.js/ally');
// load a specific module
var module = require('ally.js/query/focusable');
```

---

You can configure AMD to load the UMD:

```js
require.config({
  paths: {
    'ally.js': 'node_modules/ally.js/ally.min',
  }
});
// load ally.min.js UMD bundle
var bundle = require('ally.js');
```

*OR* you can configure AMD to load the modules:

```js
require.config({
  paths: {
    'ally.js': 'node_modules/ally.js/amd',
  }
});
// load the same structure as the UMD bundle, but from AMD modules
var bundleFromModules = require('ally.js/ally');
// load a specific module
var module = require('ally.js/query/focusable');
```

The same applies to [es6-module-loader](https://github.com/ModuleLoader/es6-module-loader/blob/master/docs/loader-config.md).

**Comment by rodneyrehm** [comment]
The approach to solve this with minimal effort would be:

1. use the [prepublish script](https://docs.npmjs.com/misc/scripts) to copy the [meta files](https://docs.npmjs.com/files/package.json#files) (`package.json`, `README.md`, `CHANGELOG.md`, `LICENSE.txt`) to `dist`
2. copy `src` to `dist/src`
3. move `dist/common/*` to `dist/*`
4. remove `"files"` from `dist/package.json`

and run [`npm publish dist`](https://docs.npmjs.com/cli/publish) to deploy the `dist` rather than the project root.

**Comment by rodneyrehm** [comment]
available to test in [1.0.0-beta.8](https://github.com/medialize/ally.js/releases/tag/1.0.0-beta.8)
